apps/src/blocksCommon.js

```diff
@@ -170,21 +208,22 @@ function installCategory(blockly) {
     },
   };
 
-  blockly.JavaScript.custom_category = function () {
-    return '\n';
-  };
+  Blockly.customBlocks.defineNewBlockGenerator(
+    blockly.JavaScript,
+    'custom_category',
+    () => {
+      return '\n';
+    }
+  );
 }
 
 function installWhenRun(blockly, skin, isK1) {
   blockly.Blocks.when_run = {
     // Block to handle event where mouse is clicked
     helpUrl: '',
     init: function () {
-      if (this.setStyle) {
-        this.setStyle('setup_blocks');
-      } else {
-        Blockly.cdoUtils.setHSV(this, 39, 1.0, 0.99);
-      }
+      Blockly.cdoUtils.setHSV(this, 39, 1.0, 0.99);
+      this.setStyle('setup_blocks');
```

Comment:
Is this logic the same because line 225 is what we want for CDO Blockly, in which case 226 is a no-op, and for Google Blockly, line 226 overrides line 255?

Reply:
Yep!